python/lsst/jointcal/jointcal.py

```diff
@@ -397,17 +406,17 @@ def _write_results(self, associations, astrom_model, photom_model, dataRefs):
             # TODO: there must be a better way to identify this ccdImage?
             name = ccdImage.Name()
             visit, ccd = name.split('_')
-            for dataRef in dataRefs:
-                calexp = dataRef.get("calexp")
-                ccdname = calexp.getDetector().getId()
-                if dataRef.dataId["visit"] == int(visit) and ccdname == int(ccd):
-                    print("Updating WCS for visit: %d, ccd%d" % (int(visit), int(ccd)))
-                    exp = afwImage.ExposureI(0, 0, tanWcs)
-                    exp.setCalib(calexp.getCalib())  # start with the original calib
-                    fluxMag0, fluxMag0Sigma = calexp.getCalib().getFluxMag0()
-                    exp.getCalib().setFluxMag0(fluxMag0*photom_model.photomFactor(ccdImage), fluxMag0Sigma)
-                    try:
-                        dataRef.put(exp, 'wcs')
-                    except pexExceptions.Exception as e:
-                        self.log.warn('Failed to write updated Wcs: ' + str(e))
-                    break
+            visit = int(visit)
+            ccd = int(ccd)
+            visit_ccd_to_dataRef[(visit, ccd)]
+            dataRef = visit_ccd_to_dataRef[(visit, ccd)]
+            calexp = dataRef.get("calexp")
```

Comment:
You could get `"calexp_md"` and use that to create the `Calib` for more time savings.